### ISSUE TITLE
Deleting object used for object-type Custom Field does not update custom_field_data, leading to invalid GraphQL output

### ISSUE THREAD

**Issue by mikkelmk** [netbox-community/netbox#12635]
### NetBox version

v3.5.1

### Python version

3.10

### Steps to Reproduce

1. Create object-type custom field with Interface as content type and Tag as object type
2. Assign a tag to this custom field on an interface
3. Delete the tag
4. Querying the custom fields on the interface through GraphQL will still show the field as being set to the id of the deleted tag

### Expected Behavior

GraphQL output like this:

```
{
  "data": {
    "interface": {
      "custom_fields": {
        "FooCustomField": null
      }
    }
  }
}
```

### Observed Behavior

GraphQL output like this:

```
{
  "data": {
    "interface": {
      "custom_fields": {
        "FooCustomField": 42
      }
    }
  }
}
```

**Note that the field will be null in both the Web interface and normal REST API** - I believe that what happens is that the underlying custom_field_data on the interface is not cleared, but that the WEB and API layers know to nullify the invalid reference

**Comment by jeremystretch** [comment]
One idea I've had on the back burner is reimplementing the custom fields feature in NetBox to utilize discrete database fields (as opposed to monolithic JSON blobs) managed via dynamic schema migrations. I haven't gotten around to drafting an FR for that yet, but it occurs to me that such a change would resolve this bug as well.

**Comment by jeremystretch** [comment]
@arthanson could you revisit this now that we've switched from Graphene to Strawberry in NetBox v4.0?

**Comment by arthanson** [comment]
This isn't blocked on GraphQL, it is blocked on Custom Field data being JSON blobs which would make this non-performant.

**Comment by jeremypng** [comment]
Agreed that this is blocked on a Housekeeper feature or converting custom fields to discrete database records.

**Comment by jeremystretch** [comment]
Is this still blocked? If so, let's identify the specific open issue(s) that are blocking it and call them out here for clarity.

**Comment by arthanson** [comment]
This is probably a high complexity issue as the custom fields are stored in JSON and trying to filter through this in a performant manner could be difficult.
